lib/components/file-manager.js

```diff
@@ -55,41 +55,37 @@ var fileManager = {
 					return console.error(err);
 				}
 
-				this.watched(function (err, watched) {
-					if (err) {
-						return console.error(err);
-					}
-					_.forEach(_.filter(_.flatten(_.values(watched)), function (file) {
-						return path.extname(file);
+				var watched = this.watched();
+				_.forEach(_.filter(_.flatten(_.values(watched)), function (file) {
```

Comment:
thx, can you bugfix release as well?

On Mon, Feb 22, 2016 at 7:08 PM, Vrolijkx <notifications@github.com> wrote:

> In lib/components/file-manager.js
> <https://github.com/Vrolijkx/platform-proxy/pull/18#discussion_r53665578>:
>
> > @@ -55,41 +55,37 @@ var fileManager = {
> >  					return console.error(err);
> >  				}
> >
> > -				this.watched(function (err, watched) {
> > -					if (err) {
> > -						return console.error(err);
> > -					}
> > -					_.forEach(_.filter(_.flatten(_.values(watched)), function (file) {
> > -						return path.extname(file);
> > +				var watched = this.watched();
> > +				_.forEach(_.filter(_.flatten(_.values(watched)), function (file) {
>
> _.chain(watched).values()
>    .flatten()
>    .filter(function (file) {
>        return path.extname(file);
>     })
>    .forEach(function(file) {
>         var cat = getCategory(file);
>         cache[cat.name].filePaths.push(file);
>    }).commit();
>
> The .commit() is essential because the chain will only be executed on a
> .commit() or .value()
> I will commit it on the branch
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Vrolijkx/platform-proxy/pull/18/files#r53665578>.
>